### PR TITLE
Add constants and change packing a little bit

### PR DESCRIPTION
**Pull request by WhiteTPoison**

### CODE REVIEW

File: BBConst.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// check clock desynchronization every X frames
+constexpr s32 ONLINE_LOCKSTEP_INTERVAL = 30;
+constexpr s32 TIMESYNC_MAX_US_OFFSET = 10000; // 60% of a frame
+
+/* ======== Debug Toggles ======== */
+//#define SYNCLOG
+//#define RANDOM_INPUTS
+
+/* ======== Utils ======== */
+#define MS_IN_FRAME (1000 / 60)
```
Comment (eliasreid) [inline BBConst.h]: I think these defines should be either constexpr vars or functions

File: FrameData.h [hunk 2]
```diff
@@ -5,6 +5,6 @@
 #pragma pack(push, 4)
 struct FrameData {
     u32 randomSeed = 0;
-    PlayerFrameData playerFrameDatas[MAX_NUM_PLAYERS];
+    PlayerFrameData playerFrameDatas[MAX_NUM_PLAYERS];
```
Comment (eliasreid) [inline FrameData.h]: assume you didn't mean to add whitespace at end of this line

File: GameSettings.h [hunk 3]
```diff
@@ -1,13 +1,15 @@
 #pragma once
+#include "BBConst.h"
```
Comment (eliasreid) [inline GameSettings.h]: Can we rename this header file to BrawlbackConstants.h

**Review by eliasreid — COMMENTED** [review]
don't have time to look through this careful and the context of how it's used in other repos, but just added a couples comments based on what's here

File: BBConst.h (hunk 1 above)
Comment (eliasreid) [reply BBConst.h]: Also, from my understanding a lot of these constants are only needed in -asm repo OR in dolphin. So I don't see why we should add them to the -common repo.

e.g. PING_DISPLAY_INTERVAL is only needed in dolphin right?

File: SavestateMemRegion.h [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+
+#pragma pack(push, 4)
+struct SavestateMemRegionInfo {
```
Comment (eliasreid) [inline SavestateMemRegion.h:5]: Is this currently being used?

File: SyncData.h [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#pragma pack(push, 4)
+struct SyncData {
```
Comment (eliasreid) [inline SyncData.h:4]: IS this currently being used?

Comment (WhiteTPoison) [reply SyncData.h:4]: Yes.

File: SavestateMemRegion.h (hunk 4 above)
Comment (WhiteTPoison) [reply SavestateMemRegion.h:5]: Yes.

File: BBConst.h (hunk 1 above)
Comment (WhiteTPoison) [reply BBConst.h]: I will more comprehensively adjust the defines, but the push I just put up addresses some of this.

**Comment by WhiteTPoison** [comment]
@eliasreid Should be good now; let me know?

**Review by eliasreid — APPROVED** [review]

**Pull request merged**